static/js/src/advantage/users/components/TableView/TableView.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import MainTable from "@canonical/react-components/dist/components/MainTable";
+
+import { Users } from "../../types";
+import { getUserRow } from "./components";
+
+export type UserRowVariant = "regular" | "editing" | "disabled";
+
+type UserInEditMode = string | null;
+type UserId = string;
+
+const getVariant = (userId: UserId, userInEditMode: UserInEditMode) => {
+  if (userInEditMode === null) {
+    return "regular";
+  } else if (userId !== userInEditMode) {
+    return "disabled";
+  } else {
+    return "editing";
+  }
+};
+
+const TableView: React.FC<{ users: Users }> = ({ users }) => {
```

Comment:
Typically we rely on the TS inference for React component definitions. So you should be able to remove `React.FC` the compiler will infer the correct types with this syntax:

```suggestion
type Props = {
  users: Users;
}

const TableView = ({ users }: Props) => {
```

Could you also make this change elsewhere in this PR plz. You can read more about how `React.FC` [is not necessary here](https://github.com/facebook/create-react-app/pull/8177)